### DISCUSSION TITLE
Unable to find tutorial data

### DISCUSSION THREAD

**Discussion by carejordan** [kschan0214/sepia#51] [General]
Hello, I am unable to find the tutorial data, as I cannot find ~/sepia_tutorial/sepia101_data/. Is it in the sepia-master folder? Thank you.

**Comment by kschan0214** [comment]
Hi Caroline,

The SEPIA101 tutorial file can be found at the following link:
https://surfdrive.surf.nl/files/index.php/s/ltV00ELb4JFDCpZ

Sorry for the confusion in the documentation website since there is no link provided there. It is because I have been thinking to revamp the tutorial using the in vivo data accompanied with the QSM recommendation paper, which is currently still working in progress (the current sepia101 data is generated using the in silico head phantom used in QSM challenge 2.0). 

The SEPIA101 tutorial data works on the latest SEPIA version (v1.2.1), but previously I noticed that there is an issue if v0.8.1 is used due to some numerical incompatibility from the in silico data, so please avoid using this particular version for the tutorial. One more thing to notice is that in Exercise 3, it asks to select an image named "Sepia_total-field.nii.gz" as the input. However, the total field is renamed to "Sepia_fieldmap.nii.gz" according to the BIDS format in the latest version. 

Please feel free to let us know if you have any questions regarding the usage or tutorial of SEPIA.

(I've moved this to the Discussion page as this topic is more relevant there)

Kwok

**Reply by carejordan** [reply]
Thank you! I was able to load the data, but when trying to complete Exercise 1 and running the Phase unwrapping, I run into an error that the "Arrays have incompatible sizes for this operation." Is this related to the numerical incompatibility mentioned? Below is the full error.

Arrays have incompatible sizes for this operation.

Error in Wrapper_EchoCombine_OptimumWeight (line 91)
        totalField = totalField + weight(:,:,:,k).*((fieldMap(:,:,:,k+1) - fieldMap(:,:,:,1))/(TE(k+1)-TE(1)));

Error in estimateTotalField (line 70)
        [totalField, N_std, headerAndExtraData] = feval(wrapper_EchoCombine_function{k},fieldMap,mask,matrixSize,voxelSize,algorParam,headerAndExtraData);

Error in UnwrapPhaseMacroIOWrapper (line 169)
[totalField,fieldmapSD,fieldmapUnwrapAllEchoes,mask] = estimateTotalField(fieldMap,mask,matrixSize,voxelSize,algorParam,headerAndExtraData);

Error in sepiaIO (line 91)
            UnwrapPhaseMacroIOWrapper(input,output,maskFullName,algorParam);

Error in sepia_config (line 23)
sepiaIO(input,output_basename,mask_filename,algorParam);

Error in run (line 91)
evalin('caller', strcat(script, ';'));

Error in sepia>PushbuttonStart_Callback (line 350)
    run(configFilename);

Related documentation
 
Error while evaluating UIControl Callback.

**Reply by kschan0214** [reply]
Hi, I think the error is related a bug in SEPIA v1.2, which is caused by one (or some) of the image dimension is an odd number. Unfortunately this is also the case with the SEPIA101 data (matrix size = 164 x **205** x **205**) :( Is that the version you are currently using (the version should be shown on the SEPIA GUI window)? If this is the case you can try to update SEPIA to the latest version 1.2.1. (https://github.com/kschan0214/sepia/releases/tag/v1.2.1) and the bug should be fixed with this version (I tested this on my computer yesterday and it did not produce any errors with v1.2.1).

Alternatively, I also have an in vivo dataset that can be shared:
https://surfdrive.surf.nl/files/index.php/s/xibjM2m9sNutBZW
However, the tutorial is not tailored for this dataset but you could at least get some ideas on the usage of SEPIA.

Let me know if any of these works!
